railties/test/application/asset_debugging_test.rb

```diff
@@ -85,12 +85,12 @@ class ::PostsController < ActionController::Base ; end
         image_submit_tag:       %r{<input type="image" src="/images/#{contents}" />}
       }
 
-      cases.each do |(view_method, tag_match)|
-        app_file "app/views/posts/index.html.erb", "<%= #{view_method} '#{contents}', skip_pipeline: true %>"
-
-        app "development"
+      app "development"
+      class ::PostsController < ActionController::Base; end
 
-        class ::PostsController < ActionController::Base ; end
+      cases.each do |(view_method, tag_match)|
+        app_file "app/views/posts/#{view_method}.html.erb", "<%= #{view_method} '#{contents}', skip_pipeline: true %>"
+        ::PostsController.define_method(:index) { render view_method }
```

Comment:
Would not this cause a warning since we are defining the same method more than once?

Reply:
You are right, I didn't notice it on dev, but it appears in the CI. You can see it with `ruby -w bin/test test/application/asset_debugging_test.rb`

The query implementation approach is easier for avoid that warning, I am going to push another commit without warnings.